README
NAME
    Dist::Zilla::Plugin::OurPkgVersion - no line insertion and does Package
    version with our

VERSION
    version 0.005001

SYNOPSIS
    in dist.ini

            [OurPkgVersion]

    in your modules

            # VERSION

DESCRIPTION
    This module was created as an alternative to
    Dist::Zilla::Plugin::PkgVersion and uses some code from that module.
    This module is designed to use a the more readable format "our $VERSION
    = $version;" as well as not change then number of lines of code in your
    files, which will keep your repository more in sync with your CPAN
    release. It also allows you slightly more freedom in how you specify
    your version.

  EXAMPLES
    in dist.ini

            ...
            version = 0.01;
            [OurPkgVersion]

    in lib/My/Module.pm

            package My::Module;
            # VERSION
            ...

    output lib/My/Module.pm

            package My::Module;
            our $VERSION = '0.01'; # VERSION
            ...

    please note that whitespace before the comment is significant so

            package My::Module;
            BEGIN {
                    # VERSION
            }
            ...

    becomes

            package My::Module;
            BEGIN {
                    our $VERSION = '0.01'; # VERSION
            }
            ...

    while

            package My::Module;
            BEGIN {
            # VERSION
            }
            ...

    becomes

            package My::Module;
            BEGIN {
            our $VERSION = '0.01'; # VERSION
            }
            ...

    you can also add additional comments to your comment

            ...
            # VERSION: generated by DZP::OurPkgVersion
            ...

    becomes

            ...
            our $VERSION = '0.1.0'; # VERSION: generated by DZP::OurPkgVersion
            ...

    Also note, the package line is not in any way significant, it will
    insert the "our $VERSION" line anywhere in the file before "# VERSION"
    as many times as you've written "# VERSION" regardless of whether or not
    inserting it there is a good idea. OurPkgVersion will not insert a
    version unless you have "# VERSION" so it is a bit more work.

    If you make a trial release, the comment will be altered to say so:

            # VERSION

    becomes

            our $VERSION = '0.01'; # TRIAL VERSION

METHODS
    munge_files
        Override the default provided by Dist::Zilla::Role::FileMunger to
        limit the number of files to search to only be modules and
        executables.

    munge_file
        tells which files to munge, see Dist::Zilla::Role::FileMunger

BUGS
    Please report any bugs or feature requests on the bugtracker website
    https://github.com/xenoterracide/dist-zilla-plugin-ourpkgversion/issues

    When submitting a bug or request, please include a test-file or a patch
    to an existing test-file that illustrates the bug or desired feature.

CONTRIBUTORS
    *   Alexandr Ciornii <alexchorny@gmail.com>

    *   Alexei Znamensky <russoz@cpan.org>

    *   Christian Walde <walde.christian@googlemail.com>

    *   Christopher J. Madsen <perl@cjmweb.net>

    *   David Golden <dagolden@cpan.org>

AUTHOR
    Caleb Cushing <xenoterracide@gmail.com>

COPYRIGHT AND LICENSE
    This software is Copyright (c) 2013 by Caleb Cushing.

    This is free software, licensed under:

      The Artistic License 2.0 (GPL Compatible)



Changes
Revision history for Dist-Zilla-Plugin-OurPkgVersion

{{$NEXT}}

0.005001  2013-12-16
	- update compile test so that it doesn't hang on windows GH #11 ( chorny )
	- add contributors plugins

0.005000  2013-09-07
	- allow comment to be #VERSION
	  - Github Issue #9
	  - Reported by Jakob Voss ( nichtich )

0.004000  2012-05-31
	- skip parsing .pod files Github Issue #6
	- Change comment to read TRIAL VERSION for trial releases
	  Github Issue #7
	  Patch by Christopher J. Madsen

0.003002  2012-01-02
	- skip if PPI unable to parse

0.003001  2012-01-01
	- Provide a better error message if PPI fails

v0.3.0    2011-08-10
	- Allow text after # VERSION comment, see docs for example
	  Github Issue #5
      Reported by David Golden

0.2.0     2011-07-31
	- Add debug logging of successfully adding versions
	  Github Issue #4
	  Patch by Christian Walde

0.1.9     2011-07-23
	- Always warn about missing # VERSION comment
	  Github Issue #3
	  Patch by David Golden

0.1.8     2011-06-01
	- Testing depends on Test::Version 0.04 or later

0.1.7     2011-05-31
	- use Test::Version::version_ok  to regression test gh issue #1

0.1.6     2011-05-09
	- version sanity checking now done the same way as PkgVersion
	- above allows for vstring, e.g 'v3.3.2', style versions,
	  Github Issue #2
	  Reported by Mike Doherty
	- use TestMania
	- add MetaJSON

0.1.5     2011-02-21
	- CRITICAL: fix bug where version number wasn't quoted. This would make
	  multipart versions (e.g. 0.1.0) not evaluate correctly, if you've used
	  a multipart version with OurPkgVersion, please re-release your module.
	- make module conform to l3 strictness of Perl::Critic
	- make changes conform to CPAN::Changes::Spec

0.1.4     2010-11-06
	- Remove AutoPrereqs from corpus, don't want to depend on that

0.1.3     2010-11-06
	- Add basic tests specific to this distribution
	- Add better error handling if no comments present
	  GitHub Issue #1
	  Reported by Stephen Clouse
	- remove cruft copied from PkgVersion that wasn't necessary

0.1.2     2010-08-07
	- All Prerequisites were not properly defined, this is corrected.

0.1.1     2010-08-07
	- override munges files to narrow which files will even be looked at

0.1.0     2010-08-03
	- initial release


Makefile.PL

use strict;
use warnings;

use 5.008;

use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "no line insertion and does Package version with our",
  "AUTHOR" => "Caleb Cushing <xenoterracide\@gmail.com>",
  "BUILD_REQUIRES" => {},
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30"
  },
  "DISTNAME" => "Dist-Zilla-Plugin-OurPkgVersion",
  "EXE_FILES" => [],
  "LICENSE" => "artistic_2",
  "NAME" => "Dist::Zilla::Plugin::OurPkgVersion",
  "PREREQ_PM" => {
    "Dist::Zilla::Role::FileFinderUser" => 0,
    "Dist::Zilla::Role::FileMunger" => 0,
    "Moose" => 0,
    "MooseX::Types::Perl" => 0,
    "PPI" => 0,
    "namespace::autoclean" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::DZil" => 0,
    "Test::More" => "0.88",
    "Test::Version" => "0.04"
  },
  "VERSION" => "0.005001",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Dist::Zilla::Role::FileFinderUser" => 0,
  "Dist::Zilla::Role::FileMunger" => 0,
  "File::Spec" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Moose" => 0,
  "MooseX::Types::Perl" => 0,
  "PPI" => 0,
  "Test::DZil" => 0,
  "Test::More" => "0.88",
  "Test::Version" => "0.04",
  "namespace::autoclean" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



